src/Drupal/DrupalKernel.php

```diff
@@ -60,6 +63,7 @@ protected function initializeContainer()
         if ($this->shouldDrushInvalidateContainer()) {
             $this->invalidateContainer();
         }
+        $this->classLoaderAddMultiplePsr4($this->getModuleNamespacesPsr4($this->getThemeFileNames()));
```

Comment:
This has to be added *somewhere* to add the theme folders to the autoloader. I did a bit of investigating and I think this is the best spot but it's the item I'm least sure about.

Reply:
Did this happen as a side effect of adding the themes to the module list before?

Does the active theme have its autoloader added?

Reply:
Didn't need it before because, one of the uses of `$this->moduleList` was propagating the autoloader.

All enabled themes get their autoloader added by this - in Drupal core, themes don't get added to the autoloader at all.

Reply:
This part is the hang up for me. I'm worried about the unpredictable consequences of Drush adding theme classes that Drupal does not add to the autoloader. Maybe this is no big deal; each theme should be uniquely namespaced, and if no one references the class, then it should not matter if it's autoloadable. I am concerned by the fact that this happens all the time when using Drush, whether or not a theme command is being executed. If this did cause a change in behavior, it could be really difficult to track down.

It might be safer to only add the autoloader when a theme command is executed. This is what the [Terminus plugin mechanism does](https://github.com/pantheon-systems/terminus/blob/1.6.1/src/Plugins/PluginAutoloadDependencies.php#L35:L46). The consequence of doing something similar for themes is that theme commands would have access to all of the classes defined in the theme, whether or not the theme was enabled, but hooks in the theme only have access to the command / hook classes in disabled themes (but should still have access to all classes in the enabled theme).

Reply:
hmm... I'm 95% sure I can simply implement that with a bit of copy/paste from there and editing. Investigating, thanks :)

Reply:
Termins has an autoload file that it can include, which is probably not going to be the case here. You may need to instantiate your own class loader. The Composer class loader works well in this capacity. I've done so successfully in the past, but at the moment I can't quite remember where, or I'd provide a link.